app/src/main/java/net/gsantner/markor/format/todotxt/TodoTxtFilter.java

```diff
@@ -180,13 +185,15 @@ public static List<Group> loadSavedFilters(final Context context) {
                 gp.keys = new ArrayList<>();
                 final JSONArray keysArray = obj.getJSONArray(KEYS);
                 for (int j = 0; j < keysArray.length(); j++) {
-                    gp.keys.add(keysArray.getString(j));
+                    final String key = keysArray.getString(i);
+                    gp.keys.add(NULL_SENTINEL.equals(key) ? null : key);
                 }
                 loadedViews.add(gp);
             }
             return loadedViews;
         } catch (JSONException e) {
             e.printStackTrace();
+            pref.edit().remove(SAVED_TODO_VIEWS).apply();
```

Comment:
Nuke saved filters if there is a parsing problem. Otherwise a problem can lead to saved filters never working (always returns empty list).